tests/test_BSplinePatch.cpp

```diff
@@ -266,15 +266,15 @@ TEST_CASE("BSplinePatch", "[nonrational][bspline_patch]")
                     .norm() == Approx(0).margin(1e-6));
 
         auto q = patch.evaluate(u_min, v_mid);
-        const auto uv0 = patch.inverse_evaluate(q, u_min, u_max, v_min, v_max);
+        const auto uv0 = std::get<0>(patch.inverse_evaluate(q, u_min, u_max, v_min, v_max));
         REQUIRE((patch.evaluate(uv0[0], uv0[1]) - q).norm() == Approx(0));
-        const auto uv1 = patch.inverse_evaluate(q, u_max - d, u_max + d, v_min, v_max);
+        const auto uv1 = std::get<0>(patch.inverse_evaluate(q, u_max - d, u_max + d, v_min, v_max));
         REQUIRE((patch.evaluate(uv1[0], uv1[1]) - q).norm() == Approx(0));
         const auto uv2 =
-            patch.inverse_evaluate(q, u_min - 2 * period - d, u_min - 2 * period + d, v_min, v_max);
+            std::get<0>(patch.inverse_evaluate(q, u_min - 2 * period - d, u_min - 2 * period + d, v_min, v_max));
         REQUIRE((patch.evaluate(uv2[0], uv2[1]) - q).norm() == Approx(0));
         const auto uv3 =
-            patch.inverse_evaluate(q, u_min - period + d, u_max - period - d, v_min, v_max);
+            std::get<0>(patch.inverse_evaluate(q, u_min - period + d, u_max - period - d, v_min, v_max));
```

Comment:
My only reservation with having a tuple returned is the more cluttered interface with std::get<0>(...), but i don't have a better idea. Changing newton-raphson() to a while loop to always converge changes functionality and isn't always desired, so maybe this is best.

Reply:
Yup, sorry about the API change.  Another option is to make either uv or the convergence flag an argout parameter.  But argout parameters are also not very nice.